dhis-2/dhis-services/dhis-service-acl/src/main/java/org/hisp/dhis/security/acl/DefaultAclService.java

```diff
@@ -94,30 +94,22 @@ public boolean isDataShareable( Class<?> klass )
         Schema schema = schemaService.getSchema( klass );
         return schema != null && schema.isDataShareable();
     }
-
+    
     @Override
     public boolean canRead( User user, IdentifiableObject object )
     {
-        if ( object == null || haveOverrideAuthority( user ) )
-        {
-            return true;
-        }
+        if ( readWriteCommonCheck( user, object ) ) return true;
 
         Schema schema = schemaService.getSchema( object.getClass() );
-
-        if ( schema == null )
-        {
-            return true;
-        }
-
+        
         if ( canAccess( user, schema.getAuthorityByType( AuthorityType.READ ) ) )
         {
             if ( object instanceof CategoryOptionCombo )
             {
                 return checkOptionComboSharingPermission( user, object, Permission.READ );
             }
 
-            if ( !schema.isShareable() || object.getUser() == null || object.getPublicAccess() == null || checkUser( user, object )
```

Comment:
What's the reason for removing `object.getUser() == null` check?

Reply:
The original `if` condition goes like:

`if ( !schema.isShareable() || object.getUser() == null || object.getPublicAccess() == null || checkUser( user, object )`

The method `checkUser(user, object)` internally checks for `object.getUser() == null` so I removed the redundant check (unless we want to keep it there to exit early).